# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWindow.cpp
# Commit: a8770c8bc4f0
# Full Hash: a8770c8bc4f02819740819a6d4415fcabab6871a
# Author: stransky <stransky@redhat.com>
# Date: 2022-12-07 04:11:18
# Regressor Bug: 1795851
# File Overlap Count: 1
# Description:
#   Bug 1795851 [Wayland] Update EGL window size/scale in moz_container_wayland_get_egl_window() r=emilio
#   
#   We need to return correct EGLWindow from moz_container_wayland_get_egl_window() with correct scale/size
#   and also keep the EGLWindow up to date. In this patch we do:
#   
# ==============================================================================

diff -r 105058a4927e -r a8770c8bc4f0 widget/gtk/nsWindow.cpp
--- a/widget/gtk/nsWindow.cpp	Tue Dec 06 15:18:31 2022 +0000
+++ b/widget/gtk/nsWindow.cpp	Tue Dec 06 15:18:32 2022 +0000
@@ -9818,9 +9818,14 @@
   if (!mContainer || !GdkIsWaylandDisplay()) {
     return;
   }
-  moz_container_wayland_egl_window_set_size(mContainer, aEGLWindowSize.width,
-                                            aEGLWindowSize.height);
-  moz_container_wayland_set_scale_factor(mContainer);
+  if (moz_container_wayland_egl_window_needs_size_update(
+          mContainer, aEGLWindowSize.ToUnknownSize(), GdkCeiledScaleFactor())) {
+    LOG("nsWindow::SetEGLNativeWindowSize() %d x %d", aEGLWindowSize.width,
+        aEGLWindowSize.height);
+    moz_container_wayland_egl_window_set_size(mContainer,
+                                              aEGLWindowSize.ToUnknownSize());
+    moz_container_wayland_set_scale_factor(mContainer);
+  }
 }
 #endif
 
